### PR TITLE
atuin: 17.2.1 -> 18.0.0

### PR DESCRIPTION
**Pull request by 0x4A6F**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by r-vdp — APPROVED** [review]
Working fine here, thanks!

**Comment by happysalada** [comment]
@0x4A6F did  you leave this in draft because of the test skips ?
this update fixes an annoying bug, that can completely lock the shell on high io pressure (so a big nixpkgs-review for example). Getting this merged would be really nice.

**Comment by 0x4A6F** [comment]
@happysalada No, just wanted to incorporate the Name-based package directories changes.
Should be good now.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -290,9 +290,9 @@ with pkgs;
 
   _0x =  callPackage ../tools/misc/0x { };
 
-  atuin = callPackage ../tools/misc/atuin {
-    inherit (darwin.apple_sdk.frameworks) AppKit Security SystemConfiguration;
-  };
+  #atuin = callPackage ../tools/misc/atuin {
```
Comment (r-vdp) [inline pkgs/top-level/all-packages.nix:293]: Why not just remove these lines? We can always recover them from the git history if needed.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -290,9 +290,9 @@ with pkgs;
...
+  #  inherit (darwin.apple_sdk.frameworks) AppKit Security SystemConfiguration;
+  #};
```
Comment (r-vdp) [inline pkgs/top-level/all-packages.nix:295]: Why not just remove these lines? We can always recover them from the git history if needed.

Comment (0x4A6F) [reply pkgs/top-level/all-packages.nix]: Yeah, that was an intermediate state.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 287492` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>atuin</li>
  </ul>
</details>

**Comment by jcollie** [comment]
There's a pretty nasty bug in 18.0.0. Please apply the patch to 18.0.0 or wait for the patch to be included in the next release.

https://github.com/atuinsh/atuin/issues/1693
https://github.com/atuinsh/atuin/pull/1694

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 287492` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>atuin</li>
  </ul>
</details>

**Comment by happysalada** [comment]
@ofborg eval

**Comment by matthiasbeyer** [comment]
What's the policy for backporting this?

The nixos option for atuin does not yet provide a way to set `package`, so updating my server side installation for nixos stable with a package from nixos unstable is not _easily_ possible yet... so naturally I'd like to see this backported...

**Comment by happysalada** [comment]
@matthiasbeyer automated backporting failed. If you make a PR though, happy to help merge it.

**Comment by matthiasbeyer** [comment]
There are some patches missing. The version on the stable channel is 17.0.1 afaik.

**Comment by matthiasbeyer** [comment]
Gonna PR in a minute! Thanks!

**Pull request merged**